Copy principal from cache if it is /different/ from current, not same

---
 src/ka-kerberos.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ka-kerberos.c b/src/ka-kerberos.c
index ee58e81..640f1ce 100644
--- a/src/ka-kerberos.c
+++ b/src/ka-kerberos.c
@@ -223,7 +223,7 @@ credentials_expiring_real (KaApplet *applet)
 
     /* copy principal from cache if any */
     if (kprincipal == NULL ||
-        krb5_principal_compare (kcontext, my_creds.client, kprincipal)) {
+        !krb5_principal_compare (kcontext, my_creds.client, kprincipal)) {
         if (kprincipal)
             krb5_free_principal (kcontext, kprincipal);
         krb5_copy_principal (kcontext, my_creds.client, &kprincipal);
-- 
1.9.1

